src/main/scala/mesosphere/marathon/upgrade/TaskReplaceActor.scala

```diff
@@ -36,11 +44,34 @@ class TaskReplaceActor(
   val periodicalRetryKills: Cancellable = context.system.scheduler.schedule(15.seconds, 15.seconds, self, RetryKills)
 
   override def preStart(): Unit = {
+    // TODO there is a race condition if this come before or after the subscribe. If it goes before
+    // TODO   the subscribe then we miss updates during that time. If it comes after, updates could
+    // TODO   still be received out of sync with this request for status.
```

Comment:
We anyway get the health check updates with a delay. It is a distributed system. The task is already healthy before that state reaches us. So we just have to make sure that we are not staying in an outdated state forever.